### DISCUSSION TITLE
⚡⚡⚡Plugin add new states and cities for WCFM (Delivery by Zone).

### DISCUSSION THREAD

**Discussion by Yan-Aleman** [inspireui/support#5656] [New Feature/Ideas Requests]
Hello. Each country has cities or states that are not in woocommerce by default. Please add compatibility with any plugin to create new states and cities (or suburbs) by default.

P/S I use . “Woo Add Custom States plugin”.
https://wordpress.org/plugins/woo-add-custom-states/

I add cities with it, then I create shipping zones in woocommerce indicating these cities or regions (with the vendors shipping method from WCFM). Then each Vendors can create their own delivery method and tariff for each delivery zone in their personal account. The advantage is that the drop-down list of cities in the user's checkout is limited to only available states or cities (the user will select the desired region anyway). And the delivery price to the city desired by the user depends only on Vendors. (settigns WCFM: one vendor = one checkout)

![1](https://github.com/inspireui/support/assets/36808995/7ad06616-d90a-4894-a378-afbe8d720168)
![2](https://github.com/inspireui/support/assets/36808995/13f490e5-6dd9-48a4-9010-66698a305619)
![image-26-02-24-03-22 копия](https://github.com/inspireui/support/assets/36808995/b2d26a1f-d516-4a25-b6a5-3b320be0a797)

**Comment by ElsaidMahmous** [comment]
it's better that mstore api can add cities

**Reply by Yan-Aleman** [reply]
Yes!

**Comment by chungxon** [comment]
Has anyone used the plugin https://wordpress.org/plugins/states-cities-and-places-for-woocommerce?

**Reply by Yan-Aleman** [reply]
We need the ability to dynamically add (independently) states, cities or satellite cities.  Because it all affects the shipping rate/price (from Vendors).  For example, food delivery (setting: Delivery Zone).  Thank you.

**Comment by DivitpatelMumbai** [comment]
Is this feature already available?

**Reply by Yan-Aleman** [reply]
No)

**Comment by mohammedkezzh** [comment]
Yes, woocommerce needs to add a list of cities or delivery zones. And automatic compatibility with flux app.
